src/www/views/servers_view/server_connection_indicator/index.ts

```diff
@@ -11,284 +11,191 @@
-      .green.INITIAL,
-      .green.DISCONNECTED,
-      .green.CONNECTING,
-      .green.RECONNECTING,
-      .green.DISCONNECTING {
-        opacity: 0;
-      }
-      .green.CONNECTED {
-        opacity: 1;
-      }
-      #small.CONNECTING,
-      #small-grey.CONNECTING,
-      #small.RECONNECTING,
-      #small-grey.RECONNECTING {
-        animation: rotate-with-pause 1.75s ease-out infinite;
-      }
-      #medium.CONNECTING,
-      #medium-grey.CONNECTING,
-      #medium.RECONNECTING,
-      #medium-grey.RECONNECTING {
-        animation: rotate-with-pause 1.75s ease-out 250ms infinite;
-      }
-      #large.CONNECTING,
-      #large-grey.CONNECTING,
-      #large.RECONNECTING,
-      #large-grey.RECONNECTING {
-        animation: rotate-with-pause 1.75s ease-out 500ms infinite;
-      }
-      #small.DISCONNECTING,
-      #small-grey.DISCONNECTING {
-        animation: rotate-backward-with-pause 1.75s ease-out infinite;
-      }
-      #medium.DISCONNECTING,
-      #medium-grey.DISCONNECTING {
-        animation: rotate-backward-with-pause 1.75s ease-out 250ms infinite;
-      }
-      #large.DISCONNECTING,
-      #large-grey.DISCONNECTING {
-        animation: rotate-backward-with-pause 1.75s ease-out 500ms infinite;
-      }
-      #small,
-      #small-grey {
-        top: 16px;
-        left: 16px;
-        height: 16px;
-        width: 16px;
-        z-index: 300;
-      }
-      #medium,
-      #medium-grey {
-        top: 8px;
-        left: 8px;
-        height: 32px;
-        width: 32px;
-        transition-delay: 250ms;
-        z-index: 200;
-      }
-      #large,
-      #large-grey,
-      #large-zero {
-        top: 0;
-        left: 0;
-        height: 48px;
-        width: 48px;
-        transition-delay: 500ms;
-        z-index: 100;
-      }
-      .expanded #small,
-      .expanded #small-grey {
-        top: 60px;
-        left: 60px;
-        height: 40px;
-        width: 40px;
+        aspect-ratio: 1;
+
+        --duration: ${ANIMATION_DURATION_MS}ms;
+        --timing-function: ease-out;
+
+        --circle-large-scale: scale(1);
+        --circle-large-delay: ${ANIMATION_DELAY_MS}ms;
+
+        --circle-medium-scale: scale(0.66);
+        --circle-medium-delay: ${ANIMATION_DELAY_MS / 2}ms;
+
+        --circle-small-scale: scale(0.33);
+        --circle-small-delay: 0ms;
+
+        --circle-initial-opacity: 0.5;
+        --circle-initial-color: grayscale(0.5);
+
+        --circle-connected-opacity: 1;
+        --circle-connected-color: grayscale(0);
+
+        --circle-disconnected-opacity: 1;
+        --circle-disconnected-color: grayscale(1);
       }
-      .expanded #medium,
-      .expanded #medium-grey {
-        top: 30px;
-        left: 30px;
-        height: 100px;
-        width: 100px;
+
+      :host,
+      .circle {
+        width: 100%;
+        height: 100%;
       }
-      .expanded #large,
-      .expanded #large-grey,
-      .expanded #large-zero {
-        top: 0;
+
+      .circle {
```

Comment:
I opted for a simplified version of BEM here that is compliant with our style standards.